apps/script-service/Controllers/ScriptController.cs

```diff
@@ -107,4 +107,38 @@ public async Task<IEnumerable<object>> RunScript(string? script, [FromBody] RunS
       return outputs;
     }
   }
+
+  [HttpPost]
+  [Route("scripts")]
+  [Authorize(AuthenticationSchemes = AdspAuthenticationSchemes.Tenant, Roles = ServiceRoles.ScriptRunner)]
+  public async Task<IEnumerable<object>> TestScript([FromBody] RunScriptRequest request)
+  {
+
+    if (request == null)
+    {
+      throw new RequestArgumentException("request body cannot be null");
+    }
+
+    if (request.ScriptDefinition == null)
+    {
+      throw new RequestArgumentException("missing script attribute in the body");
+    }
+
+    if (request.Inputs == null)
+    {
+      throw new RequestArgumentException("missing inputs attribute in the body");
+    }
+    var user = HttpContext.GetAdspUser();
+
+    Func<Task<string>> getToken = () => Task.FromResult(HttpContext.Request.Headers[HeaderNames.Authorization].First()[TOKEN_INDEX..]);
+    using (HttpContext.Benchmark("run-script-time"))
+    {
+
+      var outputs = await _luaService.RunScript(
```

Comment:
You probably need to push the test script concept down to the LuaScriptService as well, since many of the input arguments from the regular run are not applicable. Also the test run should not signal the script run events.

Reply:
Hi @tzuge, I created another function - TestScript for this endpoint. Please help me review it again. Thanks!